Rakefile

```diff
@@ -44,6 +44,58 @@ namespace :version do
 
 end
 
+class OverrideReader
+  def overrides
+    @overrides ||= {}
+  end
+  def override(name, **options)
+    overrides[name] = options
+  end
+end
+
+task :dependencies do
+  # Read the chefdk overrides file to get the list of gems and other things besides version (like source info)
+  overrides_file = File.expand_path("../omnibus/config/chefdk_overrides.rb", __FILE__)
+  puts "Reading #{overrides_file} ..."
+  reader = OverrideReader.new
+  reader.instance_eval(IO.read(overrides_file), overrides_file, 1)
+
+  # Loop through the gems in the overrides file and find the latest version of each gem
+  new_overrides = {}
+  reader.overrides.each do |name, version: nil, **options|
+    gem_name = name
+    gem_name = :'rubygems-update' if name == :'rubygems'
+    # Ask "gem list" for the latest version
+    new_version = nil
+    `gem list -r #{gem_name}`.lines.each do |line|
```

Comment:
As I mentioned before, I think you could hit http://guides.rubygems.org/rubygems-org-api/#misc-methods with a list of all the gems you want to update and do this with just one HTTP request and no shelling out.